extensions/vertx-http/runtime/src/main/java/io/quarkus/vertx/http/runtime/graal/HttpContentCompressorSubstitutions.java

```diff
@@ -0,0 +1,115 @@
+package io.quarkus.vertx.http.runtime.graal;
+
+import static io.netty.handler.codec.http.HttpHeaderValues.DEFLATE;
+import static io.netty.handler.codec.http.HttpHeaderValues.GZIP;
+import static io.netty.handler.codec.http.HttpHeaderValues.X_DEFLATE;
+import static io.netty.handler.codec.http.HttpHeaderValues.X_GZIP;
+
+import java.util.function.BooleanSupplier;
+
+import com.oracle.svm.core.annotate.Alias;
+import com.oracle.svm.core.annotate.Substitute;
+import com.oracle.svm.core.annotate.TargetClass;
+
+import io.netty.buffer.ByteBuf;
+import io.netty.channel.ChannelHandlerContext;
+import io.netty.channel.embedded.EmbeddedChannel;
+import io.netty.handler.codec.compression.ZlibWrapper;
+import io.netty.handler.codec.http2.CompressorHttp2ConnectionEncoder;
+import io.netty.handler.codec.http2.Http2Exception;
+
+public class HttpContentCompressorSubstitutions {
+
+    @TargetClass(className = "io.netty.handler.codec.compression.ZstdEncoder", onlyWith = IsZstdAbsent.class)
+    public static final class ZstdEncoderFactorySubstitution {
+
+        @Substitute
+        protected ByteBuf allocateBuffer(ChannelHandlerContext ctx, ByteBuf msg, boolean preferDirect) throws Exception {
+            throw new UnsupportedOperationException();
+        }
+
+        @Substitute
+        protected void encode(ChannelHandlerContext ctx, ByteBuf in, ByteBuf out) {
+
```

Comment:
I guess it doesn't matter much, but wouldn't it be safer to add a ` throw new UnsupportedOperationException();` line in each of these other empty methods?